pkg/assembler/graphql/schema/hasSBOM.graphql

```diff
@@ -55,9 +61,18 @@ input HasSBOMSpec {
   origin: String
   collector: String
   knownSince: Time
+  includedSoftware: [PackageOrArtifactSpec]!
+  includedDependencies: [IsDependencySpec]!
+  includedOccurrences: [IsOccurrenceSpec]!
 }
 
-"HasSBOMInputSpec is the same as HasSBOM but for mutation input."
+input HasSBOMIncludesInputSpec {
+  software: [ID!]!
```

Comment:
this is related to the issue: https://github.com/guacsec/guac/issues/1261. Based on the discussion in slack, the `pkgInputSpec` would need to contain all the IDs (for pkgType, pkgNamespace...etc) and not just the pkgVersion ID that the current `pkgInputSpec` returns

Reply:
Once #1261 is resolved, this could be modified to be:
```
input HasSBOMIncludesInputSpec {
  software: [PackageOrArtifactInputSpec!]!
  dependencies: [IsDependencyInputSpec!]!
  occurrences: [IsOccurrenceInputSpec!]!
}
```

Reply:
where only the IDs are populated for the various `InputSpec`

Reply:
I believe this has been an orthogonal discussion, but changing to the input spec would force this to either search via filtering or require only the ID to be present.  I did have another version of this code which worked solely on the input spec without the ID, however that was more involved than this one.

Reply:
yeah, it would require the ID to be present only. Not search via filtering in this case.

Reply:
I started work on issue #1261 and can update this as part of that.